CMakeLists.txt

```diff
@@ -42,47 +79,81 @@ set (lodepng_src
 add_library(zopflilib_obj OBJECT
   ${zopflilib_src}
 )
-if(BUILD_SHARED_LIBS)
+if(ZOPFLI_BUILD_SHARED)
   set_property(TARGET zopflilib_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
 #
 # libzopfli
 #
-add_library(libzopfli
+add_library(libzopfli ${zopfli_library_type}
   $<TARGET_OBJECTS:zopflilib_obj>
 )
+target_include_directories(libzopfli INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/src/zopfli")
 set_target_properties(libzopfli PROPERTIES
   OUTPUT_NAME zopfli
-  VERSION 1.0.2
-  SOVERSION 1
+  VERSION ${ZOPFLI_VERSION}
+  SOVERSION ${ZOPFLI_VERSION_MAJOR}
 )
-if(UNIX)
-  target_link_libraries(libzopfli m)
+if(UNIX AND NOT (BEOS OR HAIKU))
+  target_link_libraries(libzopfli PRIVATE m)
 endif()
 
 #
 # libzopflipng
 #
-add_library(libzopflipng
+add_library(libzopflipng ${zopfli_library_type}
   ${zopflipnglib_src}
   ${lodepng_src}
   $<TARGET_OBJECTS:zopflilib_obj>
```

Comment:
My vote for using `libzopfli` instead here to save RAM and diskspace. At least an option to avoid such duplication would be nice. Just my 2 cents.